src/content/en/2020/http2.md

```diff
@@ -4,20 +4,534 @@ part_number: IV
+
+It didn’t stop there however: over the last four years, HTTP/3 and QUIC (a new latency reducing, reliable, and secure transport protocol) have been under standards development at the IETF. There are actually two protocols that share the same name: “Google QUIC” (“gQUIC” for short), the original protocol that was designed and used by Google, and the newer IETF standardized version (IETF QUIC/QUIC). IETF QUIC was based on gQUIC, but has grown to be quite different in design and implementation. On October 21, draft 32 of IETF QUIC reached a significant milestone when it moved to [Last Call](https://mailarchive.ietf.org/arch/msg/quic/ye1LeRl7oEz898RxjE6D3koWhn0/). This is the part of the standardisation process when the working group believes they are almost finished and requests a final review.
+
+This chapter reviews the current state of HTTP/2 and gQUIC deployment, to establish how well some of the newer features of the protocol, such as prioritization and server push, have been adopted. We then look at the motivations for HTTP/3, describe the major differences between the protocol versions and discuss the potential challenges in upgrading to a UDP-based transport protocol with QUIC.
+
+### HTTP/1.0 to HTTP/2
+
+As the HTTP protocol has evolved the semantics of HTTP have stayed the same, with no changes to the HTTP methods (such as GET or POST), status codes (200, or the dreaded 404), URIs, or header fields. The differences have been the wire-encoding and use of features of  the underlying transport.
+
+HTTP/1.0, published in 1996, defined the text-based application protocol, allowing clients and servers to exchange messages in order to request resources.
+A new TCP connection was required for each request/response which introduced overhead. TCP connections use a congestion control algorithm to maximise how much data can be in-flight. This process takes time for each new connection. This "slow-start" means that not all the available bandwidth is used immediately.
+
+In 1997, HTTP/1.1 was introduced to allow TCP connection reuse by introducing 'keep-alives', aimed at reducing the total cost of connection start-ups. Over time, increasing website performance expectations led to the need for concurrency of requests. HTTP/1.1 could only request another resource after the previous response had completed. Therefore additional TCP connections had to be established, reducing the impact of the keep-alive connections and further increasing overhead.
+
+HTTP/2, published in 2015, is a binary-based protocol that introduced the concept of bidirectional streams between client and server. Using these streams, a browser can make optimal use of a single TCP connection to **multiplex** multiple HTTP requests/responses concurrently. HTTP/2 also introduced a prioritization scheme to steer this multiplexing; clients can signal a request priority that allows more important resources to be sent ahead of others.
+
```

Comment:
There are a few, very brief mentions of HTTP/3 in the text, before we describe it. It might help to just mention here something like "HTTP/3 is the next major version undergoing standardisation alongside IETF QUIC. More details at [anchor link]"

Reply:
It is described in the introduction a paragraph above ... 

>  .. HTTP/3 and QUIC (a new latency reducing, reliable, and secure transport protocol) have been under standards development in the IETF QUIC working group...

Reply:
I should l2read 😀